### ISSUE TITLE
Keras model overfitting

### ISSUE THREAD

**Issue by IvanChingLi** [IvanChingLi/tweet-classifier#5]
During training, training loss steadily decreases but validation loss makes a comeback at some point during the 10 epochs (see log/main_keras.log for details). Suggest adding regularization to main_keras.py.

**Comment by IvanChingLi** [comment]
A possibly associated issue is that the algo heavily leans towards Trump on generic sentences such as:

"how about just an everyday sentence?",
"how about just an everyday sentence? how about just an everyday sentence? how about just an everyday sentence? how about just an everyday sentence?",
"what if I just put in something random?",
"anything sounds like trump",
"wow really anything sounds like trump?"

This may be caused by the overfitting.

**Comment by IvanChingLi** [comment]
Tried regularization, dropout, and batch normalization. A combination of regularization and dropout seems to work best, although validation categorical cross entropy is still ~10x the train categorical cross entropy.
